sep/09-07-2019/README.md

```diff
@@ -0,0 +1,352 @@
+    if l1 is None and l2 is None:
+      return None
+    # add the two numbers together and make a new node with that value
+    result_node = ListNode(l1.val + l2.val)
+    # recursive call with the next value of both linked lists
+    result_node.next = self.addTwoNumbers(l1.next, l2.next)
+    return result_node
+```
+
+## Side note about the keyword `is`
+In python we have english word keywords for operators
+```python
+not # !
+or  # ||
+and # &&
+```
+However the keyword `is` is not the same as the equivalency operator `==`  
+Python's `==` is most similar to Ruby's `==`
+```ruby
+# in ruby
+a = [3, 4]
+b = [3, 4]
+a == b # true
+
+c = [4, 3]
+a = c # false
+```
+We get the same results in python
+```python
+# in python
+a = [3, 4]
+b = [3, 4]
+a == b # true
+
+c = [4, 3]
+a = c # false
+```
+
+However if we use `is` then a `is` b returns false
+```python
+# in python
+a = [3, 4]
+b = [3, 4]
+a is b # false
+```
+
+While the two arrays contains the same values in the same order they are not the same array, so if we were to modify one of them then the other array would not be modified
+```python
+a[0] = '!'
+b[1] = '!'
+
+a # ['!', 4]
+b # [3, '!']
+```
+
+In order for `a is b` to be true then we would need to assign a to b
+```python
+a = [3, 4]
+b = a
+
+a is b # true
+```
+
+## Back to the problem, we should start accounting for some edge cases
+The edge cases we should account for is
++ linked lists of different lengths
+```
+list1 = 6 -> 2 -> None
+list2 = 1 -> None
+
+result = 7 -> 2 -> None
+26 + 2 = 27
+```
++ when the values sum is greater or equal 10 then we need to carry
+```
+list1 = 6 -> 2 -> None
+list2 = 5 -> 1 -> None
+
+result = 1 -> 4 -> None
+26 + 15 = 41
+```
++ when the last two values in the linked list sums to a value greater or equal 10 then we need to add an additional node for that carry
+```
+list1 = 6 -> 2 -> None
+list2 = 1 -> 9 -> None
+
+result = 7 -> 1 -> 1 -> None
+26 + 91 = 117
+```
+
+## quick aside for ternaries in python
+In other languages we write a ternary like so  
+`variable =` (conditional) `?` (assignment if conditional is true) `:` (assignment if conditional is false)
+
+
+In python we instead write it like  
+`variable =` (assignment if conditional is true) `if` (conditional) `else` (assignment if conditional is false)
+***
+Lets tackle these edges cases in order  
+First to deal with lists of different lengths, when a node is currently at None then we can instead say the value for that node is 0
+```python
+  # Set the value to 0 if there is no number to add
+  val1 = 0 if l1 is None else l1.val
+  val2 = 0 if l2 is None else l2.val
+  # Don't proceed to next if there is no next
+  next1 = l1.next if l1 is not None else None
+  next2 = l2.next if l2 is not None else None
+  # use our new variables
+  result_node = ListNode(val1 + val2)
+  result_node.next = self.addTwoNumbers(next1, next2)
+```
+
+**Next we should handle the easier of the two carry cases**
```

Comment:
As a best practice, avoid using the bold style on an entire sentence. Only bold the few words that are critical, if any. If you really feel that the entire sentence ought to be bold, then make it the heading of a subsection.